modules/ROOT/attachments/admin-api.yaml

```diff
@@ -809,6 +809,155 @@ paths:
         200:
           description: Stop maintenance success
           content: {}
+  /v1/migrations:
+    get: 
+      summary: List all available migrations
+      operationId: list_migrations
+      responses: 
+        200:
+          description: OK
+          content: 
+            application/json:
+              {}
+      tags: 
+        - Migrations
+    put: 
+      summary: Create migration
+      operationId: add_migration
+      tags: 
+        - Migrations
+      requestBody: 
+        description: Migration configuration
+        required: true
+        content: 
+          application/json:
+            schema:                 
+              oneOf: 
+              - $ref: "#/components/schemas/inbound_migration"
+              - $ref: "#/components/schemas/outbound_migration"
+              discriminator: 
+                propertyName: migration_type
+                mapping: 
+                  inbound: "#/components/schemas/inbound_migration"
+                  outbound: "#/components/schemas/outbound_migration"
+      responses: 
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:                 
+                oneOf: 
+                - $ref: "#/components/schemas/inbound_migration"
+                - $ref: "#/components/schemas/outbound_migration"
+  /v1/migrations/{id}:
+    get:
+      summary: Get migration state
+      operationId: get_migration
+      tags: 
+        - Migrations
+      parameters: 
+        - name: id
+          in: path 
+          required: true
+          schema: 
+            type: integer
+            format: int32  
+      responses: 
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:                 
+                oneOf: 
+                - $ref: "#/components/schemas/inbound_migration"
+                - $ref: "#/components/schemas/outbound_migration"
+    post: 
+      summary: Execute migration action
+      operationId: execute_migration_action
+      tags: 
+        - Migrations
+      parameters: 
```

Comment:
There is no example or schema for this endpoint.

Reply:
@asimms41 Yes, the call will look like `curl -X POST http://localhost:9644/v1/<migration-id>/?action=prepare` with no payload, just the migration ID and the action in the URL. It's a low level API that customers shouldn't need to use as they should be using /mount and /unmount instead.

Reply:
Is there any way we can add help text/labelling?